nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/hint/AbstractGeoESHintQueryBuilder.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.elasticsearch.hint;
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST;
+import static org.elasticsearch.common.xcontent.DeprecationHandler.THROW_UNSUPPORTED_OPERATION;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.elasticsearch.api.ESHintQueryBuilder;
+
+/**
+ * @since 11.1
+ */
+public abstract class AbstractGeoESHintQueryBuilder implements ESHintQueryBuilder {
+
+    protected String[] validate(Object value, int arrayLength, String message) {
+        if (!Object[].class.isInstance(value)) {
+            throw new NuxeoException(String.format("Expected %s found %s", Object[].class, value.getClass()),
+                    SC_BAD_REQUEST);
+        }
+
+        Object[] values = Object[].class.cast(value);
+        if (values.length != arrayLength) {
+            throw new NuxeoException(String.format("Hint:%s requires %s parameters: %s", getClass().getSimpleName(),
+                    arrayLength, message), SC_BAD_REQUEST);
+        }
+
+        return Arrays.stream(values).toArray(String[]::new);
```

Comment:
You don't need a stream for that, just do `Arrays.asList(values).toArray(...)`

Reply:
I want to avoid the creation of the list and then transform to array, it's why i use the stream.

Reply:
`Arrays.asList` doesn't create a list, it returns a view of the array as a list.

Reply:
just to clarify and to have good understanding  i was thinking that asList create the list object
as below a snippet code of asList implementation where ArrayList here refer to nested class and it avoid add/remove element (UnsupportedOperationException) 
public static <T> List<T> asList(T... a) {
        return new ArrayList<>(a);
 }
thx

Reply:
Yes but this is still very cheap, it just creates one internal `ArrayList` object but that not recreate the underlying array.